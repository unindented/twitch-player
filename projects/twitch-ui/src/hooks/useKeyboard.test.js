import {
  render,
  fireEvent,
  flushEffects,
} from "@twitch-player/testing/dist/unit";
import React, { useState } from "react";
import { Text } from "react-native";
import { useKeyboard } from ".";

const SomeComponent = () => {
  const [keyCode, setKeyCode] = useState();

  useKeyboard(keyCode => {
    setKeyCode(keyCode);
  });

  return <Text>{keyCode}</Text>;
};

describe("useKeyboard", () => {
  let instance;

  beforeEach(() => {
    instance = render(<SomeComponent />);
    flushEffects();
  });

  describe("when a key is pressed", () => {
    beforeEach(() => {
      fireEvent.keyDown(document.body, { keyCode: 65 });
    });

    it("invokes the callback", () => {
      expect(instance.asFragment()).toMatchSnapshot();
    });
  });
});
