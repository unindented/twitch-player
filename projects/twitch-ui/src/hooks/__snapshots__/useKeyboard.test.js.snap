// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useKeyboard when a key is pressed invokes the callback 1`] = `
<DocumentFragment>
  <div
    class="css-text-76zvg2"
    dir="auto"
  >
    65
  </div>
</DocumentFragment>
`;
